cosmology/StructureFormation.cpp

```diff
@@ -0,0 +1,109 @@
+// StructureFormation Test
```

Comment:
Maybe it would be nice to have a few sentences describing what the simulation is about/what it's actually doing in terms of physics.